### PR TITLE
JBIDE-20763 : store 'oc' binary in ThreadLocal

### PR DESCRIPTION
**Pull request by fbricon**
Signed-off-by: Fred Bricon <fbricon@gmail.com>

### CODE REVIEW

File: src/main/java/com/openshift/restclient/OpenShiftBinary.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	 * Returns the {@link OpenShiftBinary} instance for the current {@link Thread}. 
+	 * If no instance was set via the {@link #setInstance(OpenShiftBinary)} method, 
+	 * an attempt is made to initialize a new instance from the 
+	 * "openshift.restclient.oc.location" system property.
+	 * 
+	 * @return an {@link OpenShiftBinary}
+	 * @throws LocationNotFoundException if no {@link OpenShiftBinary} can be found
+	 */
+	public static OpenShiftBinary getInstance() {
+		return threadLocalBinary.get();
```
Comment (adietish) [inline src/main/java/com/openshift/restclient/OpenShiftBinary.java]: why use a thread local if a simple static field would work?

**Comment by fbricon** [comment]
The idea is to be able to use different binaries (per OS connection), as we've seen incompatibilities already between executables and OS platforms already. Using a threadlocal makes it more thread safe than having one global static variable.

**Comment by jcantrill** [comment]
@fbricon  so i dont like the fact that now the consumer is dependent upon  a class called OpenShiftBinary; it kind of defeats the purpose for the interface abstraction where we could swap it out for a pure java impl in the future.  I suggest we move OpenShiftBinary to some form of 'Context<T>' which would allow it to be less specific and reused.  Additionally we could make the T a map that would allow passing multiple values if needed.

**Comment by fbricon** [comment]
Updated PR to use a Map<String,Object>

**Comment by jcantrill** [comment]
LGTM [merge]

**Pull request merged**